PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -1189,36 +1141,37 @@ void FillEventHistograms(T1 const& collision, T2 const& tracks, eBeforeAfter ba)
     LOGF(info, "\033[1;32m%s\033[0m", __PRETTY_FUNCTION__);
   }
 
-  // a) Common to both reconstructed and simulated;
+  // a) Common to both reconstructed and simulated:
+  // a0) collision:
   ceh_a.fEventHistograms[eNumberOfEvents][rs][ba]->Fill(0.5);
+  ceh_a.fEventHistograms[eVertex_x][rs][ba]->Fill(collision->posX());
+  ceh_a.fEventHistograms[eVertex_y][rs][ba]->Fill(collision->posY());
+  ceh_a.fEventHistograms[eVertex_z][rs][ba]->Fill(collision->posZ());
+  // a1) tracks:
 
   // b) Only reconstructed:
   if constexpr (rs == eRec) {
-    ceh_a.fEventHistograms[eTotalMultiplicity][eRec][ba]->Fill(tracks.size());
+    // b0) collision:
+    CollisionRec const* collisionRec = reinterpret_cast<CollisionRec const*>(collision);
```

Comment:
That was my first attempt, but I didn't manage to compile the code that way.

For instance, if in the function argument I have a template `T1 const* collision` and then in the function body hardcoded `collision->impactParameter()` , that didn't compile when 'collision' is aod::Collision, because that one doesn't have impactParameter() member function. 

The cast mechanism which I have implemented overcome this problem.